libsolidity/ast/Types.cpp

```diff
@@ -688,6 +687,11 @@ TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePoi
 		_other->category() != Category::Integer
 	)
 		return TypePointer();
+
+	if (auto integerType = dynamic_pointer_cast<IntegerType const>(_other)) {
+		if (integerType->isAddress()) return TypePointer();
+	}
+
 	auto commonType = Type::commonType(shared_from_this(), _other); //might be fixed point or integer
```

Comment:
I actually think that this should handle the above case already. FixedPointType and Address cannot be converted into each other and thus `commonType` is null.

Reply:
`commonType` checks if either side can be converted implicitly to the other:
- `IntegerType` allows implicit conversion to `FixedPoint` if it is not an address
- `RationalNumberType` allows implicit conversion to `FixedPoint`
- `FixedPoint` only allows implicit conversion to other configurations without precision loss of `FixedPoint`

Based on that the entire category check above the `commonType` could be removed here.